### ISSUE TITLE
pki/intermediate/set-signed missing issuer_name parameter to set issuer name easily on creation

### ISSUE THREAD

**Issue by f4z3r** [hashicorp/vault#22992]
**Is your feature request related to a problem? Please describe.**
When generating root certificates, one can provide an `issuer_name` parameter to the `pki/root/generate/:type` endpoint to provide the issuer that will get generated with a name. This is not possible on the `pki/intermediate/generate/:type` endpoint, nor on the `pki/intermediate/set-signed` endpoint. Thus one needs to perform an extra call to provide the issuer a meaningful name everytime one generates an issuer.

**Describe the solution you'd like**
Either the `pki/intermediate/generate/:type` or the `pki/intermediate/set-signed` endpoint provides an `issuer_name` parameter, allowing to set the issuer name directly as it gets created. I am a bit unsure which endpoint makes the most sense. I have the feeling the `set-signed` endpoint makes more sense though, as it is only after that call that an issuer actually gets created.

**Describe alternatives you've considered**
Rename the issuer with a separate API call after having created it.

**Explain any additional use-cases**
We have use cases where we create issuers automatically on the fly. A large number of issuers might be generated every single day. Having to rename these, and not being able to rely on the fact that our issuer names are directly set on the issuers is a bit of a pain.

**Comment by cipherboy** [comment]
@f4z3r This is because `set-signed` is not a different endpoint but [an alias of import](https://developer.hashicorp.com/vault/api-docs/secret/pki#import-ca-certificates-and-keys) in Vault 1.11+, in part because it behaves exactly the same minus disallowing the import of keys: one or more CA certificates could be imported (e.g., root CA + intermediate, or two cross-signed versions of the same intermediate, &c)...In the latter case especially, these would be two peers and could reasonably have the same name-ish (needs to be different for the API obviously, but...) 

So the harder problem is how do you solve naming for general CA import... One idea was make it a map of PEM certificate->name (assumes you know it ahead of time based on the certificates you've sent -- else you could also do two parallel arrays of the same length, one of certificates one of names), or to make `issuer_name` only work if you only specify a single certificate (do you error or warn if not?)...

In short, it is a bit harder from a what works best perspective.
